causalml/feature_selection/filters.py

```diff
@@ -0,0 +1,370 @@
+        F_test_result = pd.DataFrame({
+            'feature': X_name, # for the interaction, not the main effect
+            'method': 'F-statistic',
+            'score': F_test.fvalue[0][0], 
+            'p_value': F_test.pvalue, 
+            'misc': 'df_num: {}, df_denom: {}'.format(F_test.df_num, F_test.df_denom), 
+        }, index=[0]).reset_index(drop=True)
+
+        return F_test_result
+
+
+    def filter_F(self, data, T_name, X_name_list, y_name):
+        """
+        Rank features based on the F-statistics of the interaction.
+        """
+        all_result = pd.DataFrame()
+        for x_name_i in X_name_list: 
+            one_result = self._filter_F_one_feature(data=data,
+                T_name=T_name, X_name=x_name_i, y_name=y_name
+            )
+            all_result = pd.concat([all_result, one_result])
+
+        all_result = all_result.sort_values(by='score', ascending=False)
+        all_result['rank'] = all_result['score'].rank(ascending=False)
+
+        return all_result
+
+
+    def _filter_LR_one_feature(self, data, T_name, X_name, y_name, disp=True): 
+        """
+        Conduct LR (Likelihood Ratio) test of the interaction between treatment and one feature.
+        """
+        Y = data[y_name]
+        
+        # Restricted model
+        X_r = data[[T_name, X_name]]
+        X_r = sm.add_constant(X_r)
+        model_r = sm.Logit(Y, X_r)
+        result_r = model_r.fit(disp=disp)
+
+        # Full model (with interaction)
+        X_f = X_r.copy()
+        X_f['{}-{}'.format(T_name, X_name)] = X_f[[T_name, X_name]].product(axis=1)
+        model_f = sm.Logit(Y, X_f)
+        result_f = model_f.fit(disp=disp)
+
+        LR_stat = -2 * (result_r.llf - result_f.llf)
+        LR_df = len(result_f.params) - len(result_r.params)
+        LR_pvalue = 1 - stats.chi2.cdf(LR_stat, df=LR_df)
+
+        LR_test_result = pd.DataFrame({
+            'feature': X_name, # for the interaction, not the main effect
+            'method': 'LRT-statistic',
+            'score': LR_stat, 
+            'p_value': LR_pvalue,
+            'misc': 'df: {}'.format(LR_df), 
+        }, index=[0]).reset_index(drop=True)
+
+        return LR_test_result
+
+
+    def filter_LR(self, data, T_name, X_name_list, y_name, disp=True):
+        """
+        Rank features based on the LRT-statistics of the interaction.
+        """
+        all_result = pd.DataFrame()
+        for x_name_i in X_name_list: 
+            one_result = self._filter_LR_one_feature(data=data, 
+                T_name=T_name, X_name=x_name_i, y_name=y_name, disp=disp
+            )
+            all_result = pd.concat([all_result, one_result])
+
+        all_result = all_result.sort_values(by='score', ascending=False)
+        all_result['rank'] = all_result['score'].rank(ascending=False)
+
+        return all_result
+
+
+    # Get node summary - a function 
+    def _GetNodeSummary(self, data, 
+                        experiment_group_column='treatment_group_key', 
```

Comment:
Should we consider another default argument than 'treatment_group_key'?

Reply:
any suggestions?

Reply:
We're using 0 elsewhere in the package. Not sure that it matters that much if it is a non-public function.

Reply:
hmm, is 0 a default column name, or a default value within the experiment label column? To clarify, experiment_group_column is a column name, and data[experiment_group_column] will be the list of control / treatment labels.